xray/plot/plot.py

```diff
@@ -287,21 +288,40 @@ def _determine_extend(calc_data, vmin, vmax):
 
 def _color_palette(cmap, n_colors):
     import matplotlib.pyplot as plt
-    try:
-        from seaborn.apionly import color_palette
-        pal = color_palette(cmap, n_colors=n_colors)
-    except (TypeError, ImportError, ValueError):
-        # TypeError is raised when LinearSegmentedColormap (viridis) is used
-        # ImportError is raised when seaborn is not installed
-        # ValueError is raised when seaborn doesn't like a colormap (e.g. jet)
-        # Use homegrown solution if you don't have seaborn or are using viridis
-        if isinstance(cmap, basestring):
-            cmap = plt.get_cmap(cmap)
-
-        colors_i = np.linspace(0, 1., n_colors)
+    from matplotlib.colors import ListedColormap
+    colors_i = np.linspace(0, 1., n_colors)
+    if isinstance(cmap, (list, tuple)):
+        # we have a list of colors
+        try:
+            # first try to turn it into a palette with seaborn
+            from seaborn.apionly import color_palette
+            pal = color_palette(cmap, n_colors=n_colors)
+        except ImportError:
+            # if that fails, use matplotlib
+            # in this case, is there any difference between mpl and seaborn?
```

Comment:
Seaborn is doing something extremely similar: https://github.com/mwaskom/seaborn/blob/v0.6/seaborn/palettes.py#L404-L408